otel-extension/src/main/java/akka/actor/impl/typed/SupervisorHandleReceiveExceptionAdvice.java

```diff
@@ -0,0 +1,46 @@
+package akka.actor.impl.typed;
+
+import akka.actor.ActorContext;
+import akka.actor.ActorSystem;
+import akka.actor.typed.TypedActorContext;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.Instruments;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.impl.otel.ActorCellInstrumentationState;
+import java.util.Objects;
+import net.bytebuddy.asm.Advice;
+
+public class SupervisorHandleReceiveExceptionAdvice {
+
+  /**
+   * We need another instrumentation here as typed Actors introduced another level of supervision -
+   * implemented at [[ akka.actor.typed.internal.InterceptorImpl ]] level instead of class
+   * supervision model. To make actors_failed metric consistent we must prevent increasing this
+   * metric by instrumentation code in [[ ActorCellReceivedAdvice ]] by setting up a failed flag to
+   * true - some [[ akka.actor.typed.internal.AbstractSupervisor ]] might fall back to classic
+   * failure handle and some will recover from it and we should increase actor_failed metric in both
+   * cases.
+   */
+  @Advice.OnMethodExit(onThrowable = Throwable.class)
+  public static void exit(@Advice.Argument(0) TypedActorContext<?> typedContext) {
+    ActorContext classicContext = typedContext.asScala().classicActorContext();
+    ActorCellInstrumentationState state =
+        VirtualField.find(ActorContext.class, ActorCellInstrumentationState.class)
+            .get(classicContext);
+    Attributes attributes =
+        VirtualField.find(ActorContext.class, Attributes.class).get(classicContext);
+    Instruments instruments =
+        VirtualField.find(ActorSystem.class, Instruments.class).get(classicContext.system());
+
+    if (Objects.nonNull(instruments) && Objects.nonNull(state) && Objects.nonNull(attributes)) {
+      state.setFailed();
+      instruments.failedMessages().add(1L, attributes);
+    } else {
+      // this should never happen
+      classicContext
```

Comment:
The error message is only about actor cell state but will be printed in case: 
 - instruments are null
 - attributes are null
 - the state is null
 
 Please fix the message appropriately. Also please consider making it an "error" since this is a very unwelcome state that should not be ignored

Reply:
Actually, given the fact that the instruments, attributes and state should never be null ("this should neve happen"), I wonder if I'm not even ok with them throwing NPE, so no if/else. Should such an impossible state happen, that would mean that rapid reaction is needed. Logs will be ignored. NPE will not. :)

Reply:
(the same applies to all advice)

Reply:
bump